internal/pkg/skuba/deployments/ssh/kubelet.go

```diff
@@ -66,33 +67,30 @@ func kubeletCreateAndUploadServerCert(t *Target, data interface{}) error {
 		return errors.Wrap(err, "failure loading kubelet CA certificate authority")
 	}
 
-	host := t.target.Nodename
-	altNames := certutil.AltNames{}
-	if ip := net.ParseIP(host); ip != nil {
-		altNames.IPs = append(altNames.IPs, ip)
-	} else {
-		altNames.DNSNames = append(altNames.DNSNames, host)
-	}
-
 	// Create AltNames with defaults DNSNames/IPs
+	commonName := t.target.Nodename
+	hosts := []string{t.target.Nodename, t.target.Target, "127.0.0.1", "localhost"}
 	stdout, _, err := t.silentSsh("hostname", "-I")
 	if err != nil {
 		return err
 	}
-	for _, addr := range strings.Split(stdout, " ") {
-		if ip := net.ParseIP(addr); ip != nil {
+	hosts = append(hosts, strings.Split(stdout, " ")...)
+
+	altNames := certutil.AltNames{}
+	for _, host := range hosts {
+		if host == "" {
+			continue
+		}
+
+		if ip := net.ParseIP(host); ip != nil {
 			altNames.IPs = append(altNames.IPs, ip)
+		} else {
```

Comment:
nit: Can we add a TODO to verify that it is a valid hostname?

Reply:
Like this right? https://github.com/kubernetes/kubernetes/blob/56f60d1bcc81270fd9239ddb6b8ed83b1beb77a6/cmd/kubeadm/app/util/pkiutil/pki_helpers.go#L447-L451

Reply:
Added
https://github.com/SUSE/skuba/blob/2eaf92b2c9139b0947e3fff23a12adfbead5e6bd/internal/pkg/skuba/util/cert.go#L52-L60